### DISCUSSION TITLE
Will the ABDM support downloading from YouTube it the future?

### DISCUSSION THREAD

**Discussion by iqsalah** [amir1376/ab-download-manager#193] [Ideas]
A query about whether ABDM plans to include support for downloading content from YouTube in upcoming updates or features.

**Comment by amjadislamamjad** [comment]
Great effort. you have done a great job. just add YouTube Sniffer then your application will be marvellous

**Comment by amir1376** [comment]
> A query about whether ABDM plans to include support for downloading content from YouTube in upcoming updates or features.

Hi. about the Youtube it my not be possible because this app is hosted by GitHub but supporting normal streaming (not encrypted) is planned

**Reply by CGSudo** [reply]
many teams add Extension section. Can we add something like that too?

**Reply by Patrxgt** [reply]
> Hi. about the Youtube it my not be possible because this app is hosted by GitHub but supporting normal streaming (not encrypted) is planned

I think you could use [yt-dlp](https://github.com/yt-dlp/yt-dlp) to implement majority of video/audio downloading features without any problems with GitHub (since it's also hosted there)

**Reply by Nirzak** [reply]
> > A query about whether ABDM plans to include support for downloading content from YouTube in upcoming updates or features.
> 
> Hi. about the Youtube it my not be possible because this app is hosted by GitHub but supporting normal streaming (not encrypted) is planned

If youtube downloading tool lik yt dlp can exist on github then your project also bro.

**Comment by ebrahim-zarei** [comment]
Is there generally a way to download from YouTube with this software?

**Comment by alijasimy** [comment]
In my opinion, the only weakness this program has over Download Manager is that it doesn't download from multimedia sites like YouTube or Aparat, etc. If they add this, I dare say that everyone will install this program because it has no other advantages over the original Download Manager.

**Comment by doudoud82** [comment]
Using yt-dlp as a backend to support multimedia would be a huge improvement.
[yt-dlp](https://github.com/yt-dlp/yt-dlp) supports a huge number of sites to begin with, and they tend to remove support for websites that mainly do piracy.
A way to implement it can be done by checking if the users are in (supported websites or if a website has embeds or Iframes to supported websites, assuming yt-dlp have a list of hosts to cross-reference) or (just for starter pass the current website URL for yt-dlp. and if they already support finding embeds in websites then that less work), then pass the URL to the AB download manager, and then from there invoke yt-dlp to grab a download URL. All settings for yt-dlp can be set in the AB download manager (quality, etc).
A lot of applications already use yt-dlp for streaming content, and they can serve as a learning point of how to utilise yt-dlp, such as:
- MPV
- MPC-HC
- and many others
The functionality can be disabled for users who don't have yt-dlp installed/available in PATH (the App should inform the extension to enable multimedia support or not), as for yt-dlp, I think it's best to let the users install it themselves for the feature, just have enough documentation to hint at the functionality, assuming that the setting in AD download manager won't already be a good enough hint.

P.S: I think IDM is an example of an application that doesn't use yt-dlp since they can grab way more multimedia content from my personal experience than yt-dlp, but they are closed source and have years’ worth of development to reach their current status quo and not to mention their extension isn't in the Chrome store but gets side-loaded when installing IDM.

**Comment by amir1376** [comment]
> In my opinion, the only weakness this program has over Download Manager is that it doesn't download from multimedia sites like YouTube or Aparat, etc. If they add this, I dare say that everyone will install this program because it has no other advantages over the original Download Manager.


Hi. I've implemented this feature in v1.7.0. 
however it doesn't work with encrypted streaming websites like Youtube (which is intentional due to copyright and legal concerns as I mentioned earlier)

**Reply by Nou4r** [reply]
What kind of crack are you smoking to think that downloading from youtube is illegal and will get you in trouble with github? Do you have down syndrome on top of your albino complexity? I beg your pardon.

**Reply by petesimon** [reply]
> What kind of crack are you smoking to think that downloading from youtube is illegal and will get you in trouble with github? Do you have down syndrome on top of your albino complexity? I beg your pardon.

Stop the hate speech. and read the terms and conditions on Youtube's site. https://www.youtube.com/static?template=terms

<img width="652" height="414" alt="image" src="https://github.com/user-attachments/assets/bf72f4d6-6619-4cf1-a309-690d5ccd8289" />

**Comment by Brandglider** [comment]
Please Add [yt-dlp], it already on github with many other projects.

Please, have someone fork it if possible. This is the only thing that letting this downloder feel short

**Comment by sardar-admin** [comment]
That would make this app the ultimate downloader. Lot of users are waiting for that feature.
Anyway I thank and respect the developers for making such a great opensource application.

**Comment by kerollosy** [comment]
I made a fork that has yt-dlp support: https://github.com/kerollosy/ab-download-manager
paired with the extension as well: https://github.com/kerollosy/ab-download-manager-browser-integration

**Reply by LiveOnlyFans** [reply]
nice try

**Comment by M0VO** [comment]
is there any development for Youtube Downloader?

**Reply by alijasimy** [reply]
https://github.com/kerollosy/ab-download-manager

With this version, you can download from YouTube.

thanks @kerollosy
